styx-common/src/main/java/com/spotify/styx/storage/DatastoreStorage.java

```diff
@@ -394,17 +493,6 @@ public void updateNextNaturalTrigger(WorkflowId workflowId, TriggerInstantSpec t
         .collect(toMap(RunState::workflowInstance, Function.identity()));
   }
```

Comment:
This is just one of the methods where we used to do a weakly consistent query. E.g. `readActiveStates(String componentId)` also is used by `styx ls` and will return an inconsistent view... Maybe cool to fix in another pr.

Reply:
Caching `readActiveStates` (or similar state) might reintroduce the problem of not being able to read writes.

E.g. a user triggers a wfi and then does a read/listing that returns data from the cached `readActiveStates`. The user might then be surprised that their newly triggered wfi is not included in the response.